c2corg_ui/views/document.py

```diff
@@ -413,20 +421,43 @@ def _get_api_cache_key_from_etag(self, etag):
     def _get_response(self, page_html):
         return get_response(self.request, page_html)
 
+    def _redirect(self, id, lang, slug=None):
+        location = ''
+        if slug is None:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view_id_lang', id=id, lang=lang)
+        else:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view', id=id, lang=lang, slug=slug)
+        raise HTTPMovedPermanently(location=location)
```

Comment:
> It's better if we use HTTPFound (302) instead of HTTPMovedPermanently (301).

The goal of using 301 is to make it clear to search engines that the redirected URL should be used instead of the original URL ("permanent redirection"). Not sure 302 would be understood the same way by search engines.

https://moz.com/learn/seo/http-status-codes says that
> Use 301 redirects rather than 302 redirects when redirecting URLs on a site to ensure that link juice (ranking power) is passed between the redirecting web pages.

and https://moz.com/learn/seo/redirection
> 301, "Moved Permanently"—recommended for SEO